tests/test_naive_bayes.py

```diff
@@ -0,0 +1,394 @@
+
+	# make sure univariate distributions are working
+	assert_almost_equal( normal.log_probability( 11 ), -6.112085713764219 )
+	assert_almost_equal( normal.log_probability( 9 ), -3.612085713764219 )
+	assert_almost_equal( normal.log_probability( 7 ), -2.112085713764219 )
+	assert_almost_equal( normal.log_probability( 5 ), -1.612085713764219 )
+	assert_almost_equal( normal.log_probability( 3 ), -2.112085713764219 )
+	assert_almost_equal( normal.log_probability( 1 ), -3.612085713764219 )
+	assert_almost_equal( normal.log_probability( -1 ), -6.112085713764219 )
+
+	assert_almost_equal( uniform.log_probability( 11 ), -float('inf') )
+	assert_almost_equal( uniform.log_probability( 10 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( 5 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( 0 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( -1 ), -float('inf') )
+	
+	# create multivariate distributions, test values of distributions
+	multi = MultivariateGaussianDistribution( means=[ 5, 5 ], covariance=[[ 2, 0 ], [ 0, 2 ]] )
+	indie = IndependentComponentsDistribution( distributions=[ UniformDistribution(0, 10), UniformDistribution(0, 10) ])
+
+	# make sure multivariate distributions are working
+	assert_almost_equal( multi.log_probability([ 11, 11 ]), -20.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 9, 9 ]), -10.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 7, 7 ]), -4.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 5, 5 ]), -2.5310242469699453 )
+	assert_almost_equal( multi.log_probability([ 3, 3 ]), -4.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 1, 1 ]), -10.531024246969945 )
+	assert_almost_equal( multi.log_probability([ -1, -1 ]), -20.531024246969945 )
+
+	assert_almost_equal( indie.log_probability([ 11, 11 ]), -float('inf') )
+	assert_almost_equal( indie.log_probability([ 10, 10 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ 5, 5 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ 0, 0 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ -1, -1 ]), -float('inf') )
+
+	# create simple hmms, test values of hmms
+	rigged = State( DiscreteDistribution({ 'H': 0.8, 'T': 0.2 }) )
+	unrigged = State( DiscreteDistribution({ 'H': 0.5, 'T':0.5 }) )
+	
+	dumb = HiddenMarkovModel()
+	dumb.start = rigged
+	dumb.add_transition( rigged, rigged, 1 )
+	dumb.bake()
+
+	fair = HiddenMarkovModel()
+	fair.start = unrigged
+	fair.add_transition( unrigged, unrigged, 1 )
+	fair.bake()
+
+	smart = HiddenMarkovModel()
+	smart.add_transition( smart.start, unrigged, 0.5 )
+	smart.add_transition( smart.start, rigged, 0.5 )
+	smart.add_transition( rigged, rigged, 0.5 )
+	smart.add_transition( rigged, unrigged, 0.5 )
+	smart.add_transition( unrigged, rigged, 0.5 )
+	smart.add_transition( unrigged, unrigged, 0.5 )
+	smart.bake()
+
+	# make sure hmm's are working
```

Comment:
basically the setup function just produces the models. no testing should happen in them. the testing happens in their own functions later on